app/src/test/java/com/android/sample/model/map/NominatimLocationRepositoryTest.kt

```diff
@@ -124,6 +136,35 @@ class NominatimLocationRepositoryTest {
     assertTrue(failureCalled)
   }
 
```

Comment:
This is a well-written test that successfully verifies the failure handling logic of getCurrentLocation. The approach is thorough, and the assertions are well-suited to ensure that the behavior meets expectations. Great job!